### PR TITLE
Add system thread attribute to SearchTransportService metrics

### PR DESCRIPTION
**Pull request by carlosdelest**
Adds a `system_thread` attribute to metrics collected in the `SearchTransportService`.

This will allow to remove system indices from search metrics, to avoid polluting metrics with internal ES operations.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java [hunk 1]
```diff
@@ -666,7 +667,12 @@ private static <Request extends TransportRequest> TransportRequestHandler<Reques
     ) {
         var threadPool = transportService.getThreadPool();
         var latencies = searchTransportMetrics.getActionLatencies();
-        Map<String, Object> attributes = Map.of(ACTION_ATTRIBUTE_NAME, actionQualifier);
+        Map<String, Object> attributes = Map.of(
+            ACTION_ATTRIBUTE_NAME,
+            actionQualifier,
+            SYSTEM_THREAD_ATTRIBUTE_NAME,
+            ThreadPool.isSystemThreadPool()
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java:674]: At a quick glance, it seems we're not on the correct thread when this function is invoked. The forking in the search thread pool occurs later, so this should likely be moved to the lambda defined below?

Comment (carlosdelest) [reply server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java]: 🤦 Correct - fixed in ea6a2ae837e87abade62c7df8e651394fc06e750.

Would there be a simple way to test this? Is there a way to ensure that the search is going to be forked to the corresponding executor?

Comment (jimczi) [reply server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java]: Looking at the callers of `instrumentedHandler` it looks like we're not measuring the right thing. Most of the function call referenced in the instrumented handler lambdas are asynchronous and will delegate the actual execution of the search in a different thread. 
See for instance [here](https://github.com/elastic/elasticsearch/blob/97a724e499b6aae5c819f90d0c14a84a746a6ca3/server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java#L526) where we call `SearchService#executeQueryPhase`. However this function is not synchronous so I think we're just measuring the time it takes to move the actual execution on a different thread? cc @javanna

**Comment by carlosdelest** [comment]
Closed as this will be worked on in https://github.com/elastic/elasticsearch/pull/113194

**Pull request closed without merging**